lib/torrent.js

```diff
@@ -473,19 +473,14 @@ class Torrent extends EventEmitter {
 
     this._rarityMap = new RarityMap(this)
 
+    this.files = this.files.map(file => new File(this, file))
     let rawStore = this._preloadedStore
     if (!rawStore) {
       rawStore = new this._store(this.pieceLength, {
-        // opts.torrent is deprecated (replaced by the name property).
-        // it doesn't appear to be used by current versions of any stores on npm.
-        torrent: {
-          infoHash: this.infoHash
-        },
-        files: this.files.map(file => ({
-          path: path.join(this.path, file.path),
-          length: file.length,
-          offset: file.offset
-        })),
+        torrent: this,
+        path: this.path, // for stores that don't care about torrent
+        // files[n].path.startsWith(path) === true
+        files: this.files,
```

Comment:
That breaks compat with a lot more than just fs-chunk-store.

This could have implications for any store who used this property as a means to identify a file.   If you change this, metadata stores may have used to keep track of files will now be incorrect.  This could lead to orphaned files, cause new files to download when existing data already existed, or data to be lost until the stores are updated...  

I think change is far worse than the other.